tiatoolbox/visualization/tileserver.py

```diff
@@ -47,7 +51,39 @@ def __init__(self, title: str, layers: Dict[str, WSIReader]) -> None:
             static_folder=data._local_sample_path(Path("visualization") / "static"),
         )
         self.tia_title = title
-        self.tia_layers = layers
+
+        # generic layer names if none provided
+        if isinstance(layers, list):
+            layers = {f"layer-{i}": p for i, p in enumerate(layers)}
+        # set up the layer dict
+        layer_def = {}
+        meta = None
+        for i, key in enumerate(layers.keys()):
+            layer = layers[key]
+
+            if isinstance(layer, (str, Path)):
+                p = Path(layer)
+                if p.suffix in [".jpg", ".png"]:
+                    # assume its a low-res heatmap
+                    layer = Image.open(p)
+                    layer = np.array(layer)
+                else:
+                    layer = WSIReader.open(p)
+
+            if isinstance(layer, np.ndarray):
+                if len(layer.shape) == 2:
+                    # single channel, make into rgb with colormap
+                    c_map = cm.get_cmap("jet")
```

Comment:
It would be nice to be able to change the colormap. I'm personally not a big fan of jet as a hardcoded colormap because it is not perceptually uniform. The luminance does not correspond to the value, hence it is not good for colour blind users. If we must have a hardcoded map or a default I suggest "viridis" (the matplotlib default) or "magma". See here for more information:
- https://matplotlib.org/3.5.0/tutorials/colors/colormaps.html
- https://jakevdp.github.io/blog/2014/10/16/how-bad-is-your-colormap/

Reply:
changed the default to viridis, and can choose the colourmap in colourise function, though there is no provision for choosing colourmap in cli at the moment. Also addressed the other code maintainablility/readability issues above